spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/AbstractRabbitListenerContainerFactoryConfigurer.java

```diff
@@ -107,9 +118,13 @@ protected void configure(T factory, ConnectionFactory connectionFactory,
 			RetryInterceptorBuilder<?> builder = (retryConfig.isStateless()
 					? RetryInterceptorBuilder.stateless()
 					: RetryInterceptorBuilder.stateful());
-			builder.maxAttempts(retryConfig.getMaxAttempts());
-			builder.backOffOptions(retryConfig.getInitialInterval(),
-					retryConfig.getMultiplier(), retryConfig.getMaxInterval());
+			RabbitListenerRetryTemplateConfigurer retryConfigurer = this.retryTemplateConfigurer;
+			if (retryConfigurer == null) {
+				retryConfigurer = new RabbitListenerRetryTemplateConfigurer();
+			}
+			RetryTemplate template = new RetryTemplate();
+			retryConfigurer.configure(template, retryConfig);
+			builder.retryOperations(template);
```

Comment:
This code doesn't is not equivalent right? `RabbitListenerRetryTemplateConfigurer()`  does more than what you removed as far as I see.

Reply:
It's exactly equivalent; just that the old code used a shortcut on the builder. In any case; I'll refactor to a customizer and this will be moot.